packages/server/src/api/index.ts

```diff
@@ -66,9 +71,13 @@ if (apiEnabled()) {
     )
     .use(pro.licensing())
     .use(currentApp)
-    .use(auth.auditLog)
-    .use(migrations)
-    .use(cleanup)
+
+  // Add CSP as soon as possible - depends on licensing and currentApp
+  if (!coreEnv.DISABLE_CONTENT_SECURITY_POLICY) {
+    router.use(middleware.csp)
+  }
```

Comment:
The CSP middleware has been removed from the base koa file and instead spliced in here, after the licensing and current app middle which this feature depends on.